packages/sdk/ios/Sources/DataModels/PrimerSettings+Extensions.swift

```diff
@@ -43,39 +43,65 @@ extension PrimerSettings {
-                    if let additionalFieldsStrings = rnShippingOptions["additionalShippingContactFields"] as? [String] {
-                        additionalShippingContactFields = additionalFieldsStrings.compactMap { fieldStr in
+                    var shippingContactFields: [PrimerApplePayOptions.RequiredContactField]? = nil
+                    if let requiredFieldsStrings = rnShippingOptions["shippingContactFields"] as? [String] {
+                        shippingContactFields = requiredFieldsStrings.compactMap { fieldStr in
                             switch fieldStr {
                             case "name":
                                 return .name
                             case "emailAddress":
                                 return .emailAddress
                             case "phoneNumber":
                                 return .phoneNumber
+                            case "postalAddress":
+                                return .postalAddress
                             default:
                                 return nil // Ignore unknown values or handle accordingly
                             }
                         }
                     }
 
                     shippingOptions = PrimerApplePayOptions.ShippingOptions(
-                        isCaptureShippingAddressEnabled: isCaptureShippingAddressEnabled,
-                        additionalShippingContactFields: additionalShippingContactFields,
+                        shippingContactFields: shippingContactFields,
                         requireShippingMethod: requireShippingMethod
                     )
                 }
 
+                var billingOptions: PrimerApplePayOptions.BillingOptions? = nil
+                if let rnBillingOptions = rnApplePayOptions["billingOptions"] as? [String: Any] {
+                    var requiredBillingContactFields: [PrimerApplePayOptions.RequiredContactField]? = nil
+                    if let requiredFieldsStrings = rnBillingOptions["requiredBillingContactFields"] as? [String] {
+                        requiredBillingContactFields = requiredFieldsStrings.compactMap { fieldStr in
+                            switch fieldStr {
+                            case "name":
+                                return .name
+                            case "emailAddress":
+                                return .emailAddress
+                            case "phoneNumber":
+                                return .phoneNumber
+                            case "postalAddress":
+                                return .postalAddress
+                            default:
+                                return nil // Ignore unknown values or handle accordingly
+                            }
+                        }
+                    }
+
+                    billingOptions = PrimerApplePayOptions.BillingOptions(
+                        requiredBillingContactFields: requiredBillingContactFields
```

Comment:
[question]: do we need to filter `null` values?

Reply:
We are already doing that by using `compactMap`, correct? @semirp

Reply:
cool. didn't know about how it works